src/Symfony/Bundle/FrameworkBundle/Command/CachePoolClearCommand.php

```diff
@@ -75,7 +81,12 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
         foreach ($pools as $id => $pool) {
             $io->comment(sprintf('Clearing cache pool: <info>%s</info>', $id));
-            $pool->clear();
+
+            if ($pool instanceof CacheItemPoolInterface) {
+                $pool->clear();
+            } else {
+                $globalClearer->clearPool($id);
```

Comment:
Either we keep `hasPool()/clearPool()` or we get all the pools and let the caller doing the filtering logic as you propose, it's debatable but I would prefer keep the clearer being a clearer rather than an accessor. As you prefer :)